doc/rst/language/evolution.rst

```diff
@@ -11,9 +11,111 @@ papers that predated the changes.
+that behave differently due to these changes.
+
+See also:
+
+ * The `1.27 overload resolution rules <https://chapel-lang.org/docs/1.27/language/spec/procedures.html#function-resolution>`_
+ * The `1.28 overload resolution rules <https://chapel-lang.org/docs/1.28/language/spec/procedures.html#function-resolution>`_
+
+Param Expression Behavior
+*************************
+
+Some expressions consisting of mixed-type literal or ``param`` values now
+have different behavior. For example:
+
+.. code-block:: chapel
+
+    var x = 1:int(8) + 2; // what is the type of `x` ?
+
+Note in this example that the literal ``2`` is a ``param`` with type
+``int(64)`` and that ``1:int(8)`` is a ``param`` with type ``int(8)``.
+
+In 1.27, this program would output ``int(8)``, because the overload
+resolution rules would favor the ``+`` overload using the type of the
+non-default-sized ``param``. The result is that in 1.27, ``x`` had type
+``int(8)``.
+
+In 1.28, the rules are simpler and a closer match to the corresponding
+case with regular variables (``myInt8 + myInt64``). There is no longer
+any special behavior for non-default-sized ``param``. As a result, the
+value ``x`` now has type ``int(64)``.
+
+For similar reasons, the type of ``nI`` in the following code is now
+``int(64)`` where previously it was ``int(32)``:
+
+.. code-block:: chapel
+
+    const nI = ((-2):int(32))**53;
+
+A similar change can also appear with range literals that use mixed type
+``param`` lower and upper bounds. The following range construction also
+makes use of the new implicit conversion from ``int(t)`` to ``uint(t))``:
+
+.. code-block:: chapel
+
+    var r8 = 1:int(8)..100:uint(8);
+    writeln(r8.type:string);
+
+In 1.27, this would generate a range with index type ``int(16)``. In
+1.28, it produces a range with index type ``uint(8)``.
+
+Speaking of range literal construction, a range like ``1:int(8)..10``
+still produces an ``int(8)`` range in 1.28. However, as we have
+discussed, something like ``1:int(8) + 10`` would result in an
+``int(64)``. For now, the range implementation has been adjusted to
+preserve the old behavior specifically for the ``..`` operator. However,
+this may change in a future release.
+
+Change for some mixed int/uint overloads
+****************************************
+
+This example shows a change in behavior for two overloads where one is
+int and the other is uint:
+
+.. code-block:: chapel
+
+    proc dbm(a:int(8))   { writeln("dbm int8"); }
+    proc dbm(a:uint(64)) { writeln("dbm uint64"); }
+
+    dbm(42:int(64));
+
+Previous to 1.28, this program would call the ``int(8)`` version of the
+function. It can do that because the compiler knows that the ``param``
+value ``42`` will fit into an ``int(8)``. Such a conversion is called a
+``param`` narrowing conversion. However, in 1.28, this function now calls
+the ``uint(64)`` version of the function. The main reason for this is
+that the 1.28 rules prefer to not do ``param`` narrowing conversion.
```

Comment:
I wonder if someone might read this and conclude that the narrowing behavior is no longer supported.

Should we say: "... prefer to not do ``param`` narrowing conversion when an ``int`` to ``uint`` conversion is available``, or something like that?